### DISCUSSION TITLE
Will you remove CsWebkit and CsScript?

### DISCUSSION THREAD

**Discussion by ghost** [RolandHughes/copperspice#3] [General]
They are not usable for a while. This is extracted from the KitchenSink demo:

```
# webkit temporarily removed, pending overflow issue resolution in CsWebKit
```

**Comment by RolandHughes** [comment]
Don't have plans either way right now.

Rip out all remaining Qt Declarative and QML supporting code

That step of the roadmap will determine what happens. That's the last place portions of those directory trees still exist.

**Comment by RolandHughes** [comment]
Preface: I tend to write "this is how the sausage is made" type emails and design documentation because that helps my brain work through the options. My clients like this because they never have to question _how_ I made a decision of biblical shift.

Yes, it is mostly going to be gone. I ran into some serious code issues doing some renames. There are obviously huge chunks of dead code in that tree because I found things that could not have ever compiled.

My knee-jerk was to look at RDK wrapper. Might still do that, but [Servo](https://servo.org/) is going to be my first dumpster dive through the documentation.

This [RDK Wiki](https://wiki.rdkcentral.com/display/RDK/rdkbrowser) is dramatically out of date. QtWebkit was killed off. There is one guy jousting the windmills trying to keep it alive. I'm busy with harvest right now and have _not_ had time to dig into the code, but I believe the QtWebKit code in CopperSpice is the same that was in 4.8.x which would make it massively out of date. That would pre-date [Webkit2](https://trac.webkit.org/wiki/WebKit2).

The QtWebEngine went with the Blink stuff from Google/Chromium. I don't trust Google any further than I can throw a [1967 Ford Fairlane](https://en.wikipedia.org/wiki/Ford_Fairlane_(Americas)). They will _always_ bury spyware in their code.

I am definitely not impressed with [Ephiphany](https://gitlab.gnome.org/GNOME/epiphany) and that is about the only non-Apple browser using the library.

Servo will give us both WebGL and WebGPU capabilities out of the box . . . assuming it has or can be made to have a C++ callable API.

Take your Brave or Vivaldi browser and visit [here](https://get.webgl.org/). See the spinning box.

The primary goal of this project is to support embedded which is also a primary goal of the Servo project. Despite the fact that project is written in Rust, are objectives align. The primary goal of WebKit is whatever Apple wants it to be and Apple isn't a player in the embedded systems/medical device world.

I have 7 forecasted days of sunshine for soybean harvesting so won't really be contributing code wise to this project during that time. There will just be some late night dumpster diving "this is how the sausage is made" notes and documentation.

I'm about 90% done with the macro and method renames in the code. Once that is done I will check it into branch and someone can opt to fork the branch to rename the files starting with Cs cs_ or some combination thereof. In my mind the documentation and process is almost completely assembled for the Armv8 cross compiling and Docker building.

Once that is done the branch will be merged with default/main.

Post GLFW migration is when I think a good time will be for the Final project name and 100% migration to SourceForge. That's the site for professional projects.

**Reply by ghost** [reply]
I will go to SourceForge with you. I'm a user of JEdit and this project is also on SourceForge.
